cmake/api_macros.cmake

```diff
@@ -8,11 +8,13 @@ macro(OPENDDS_GET_SOURCES_AND_OPTIONS
   idl_prefix
   libs
   tao_options
-  opendds_options)
+  opendds_options
+  suppress_typecodes)
 
   set(_options_n
     PUBLIC PRIVATE INTERFACE
-    TAO_IDL_OPTIONS OPENDDS_IDL_OPTIONS)
+    TAO_IDL_OPTIONS OPENDDS_IDL_OPTIONS
+    OPENDDS_SUPPRESS_TYPECODES)
```

Comment:
The `tao_idl` options include both `-Sa` and `-St`, so both anys and typecodes are involved.  The MPC feature is `dds_suppress_anys` although it impacts both.  One could decide to have typecodes without anys (but not the other way around) but maybe this could be handled by directly adding the command line parameters to the rules for the code generators.